### PR TITLE
Eliminate the redundant control flow

### PR DESCRIPTION
**Pull request by Lewuathe**
The optimizer module has several unnecessary `if` statements inside. We can simplify such control flows by eliminating them.

It relates to https://github.com/prestosql/presto/pull/4300.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/sql/planner/Symbol.java [hunk 1]
```diff
@@ -68,11 +68,7 @@ public boolean equals(Object o)
 
         Symbol symbol = (Symbol) o;
 
-        if (!name.equals(symbol.name)) {
-            return false;
-        }
-
-        return true;
+        return name.equals(symbol.name);
```
Comment (wendigo) [inline presto-main/src/main/java/io/prestosql/sql/planner/Symbol.java:71]: 👍

File: presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java [hunk 2]
```diff
@@ -195,9 +195,7 @@ public static LocalDynamicFilterConsumer create(JoinNode planNode, List<Type> bu
     private static boolean isFilterAboveTableScan(PlanNode node)
     {
         if (node instanceof FilterNode) {
-            if (((FilterNode) node).getSource() instanceof TableScanNode) {
-                return true;
-            }
+            return ((FilterNode) node).getSource() instanceof TableScanNode;
         }
         return false;
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java]: return node instanceof FilterNode && ((FilterNode) node).getSource() instanceof TableScanNode;

Comment (Lewuathe) [reply presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java]: I found it has been done in https://github.com/prestosql/presto/pull/4300

**Comment by losipiuk** [comment]
@Lewuathe Please rebase

**Comment by Lewuathe** [comment]
@losipiuk Thanks. I rebased.

**Comment by sopel39** [comment]
merged, thanks!

**Pull request merged**